rskj-core/src/main/java/co/rsk/core/bc/TransactionPoolImpl.java

```diff
@@ -330,34 +360,48 @@ public synchronized void removeObsoleteTransactions(long timeSeconds) {
     private void removeTransactionList(List<Keccak256> toremove) {
         for (Keccak256 key : toremove) {
             pendingTransactions.remove(key);
-            wireTransactions.remove(key);
+            queuedTransactions.remove(key);
+
+            Transaction tx = queuedTransactions.get(key);
+
+            if (tx != null) {
+                queuedTransactionsByAccount.get(tx.getSender()).remove(tx);
```

Comment:
If this is the last txs in `queuedTransactionsByAccount.get(tx.getSender())` we can remove the list from the map.